script.js

```diff
@@ -97,45 +125,82 @@ function setPlatform(platform) {
+        portField.value = newPort;
+    }
+}
+
+function generateRegionInfo(name, fqdn, port, protocol) {
     const regions = [
         // Add default regions so they also show up in the menu
         {
-            $type: "DnsRegionInfo, Assembly-CSharp",
-            Fqdn: "na.mm.among.us",
-            DefaultIp: "50.116.1.42",
-            Port: 22023,
+            $type: "StaticHttpRegionInfo, Assembly-CSharp",
             Name: "North America",
+            PingServer: "matchmaker.among.us",
+            Servers: [
+                {
+                    Name: "Http-1",
+                    Ip: "https://matchmaker.among.us",
+                    Port: 443,
+                    UseDtls: true,
+                    Players: 0,
+                    ConnectionFailures: 0,
+                },
+            ],
             TranslateName: 289,
-            UseDtls: true,
         },
         {
-            $type: "DnsRegionInfo, Assembly-CSharp",
-            Fqdn: "eu.mm.among.us",
-            DefaultIp: "172.105.251.170",
-            Port: 22023,
+            $type: "StaticHttpRegionInfo, Assembly-CSharp",
             Name: "Europe",
+            PingServer: "matchmaker-eu.among.us",
+            Servers: [
+                {
+                    Name: "Http-1",
+                    Ip: "https://matchmaker-eu.among.us",
+                    Port: 443,
+                    UseDtls: true,
+                    Players: 0,
+                    ConnectionFailures: 0,
+                },
+            ],
             TranslateName: 290,
-            UseDtls: true,
         },
         {
-            $type: "DnsRegionInfo, Assembly-CSharp",
-            Fqdn: "as.mm.among.us",
-            DefaultIp: "139.162.111.196",
-            Port: 22023,
+            $type: "StaticHttpRegionInfo, Assembly-CSharp",
             Name: "Asia",
+            PingServer: "matchmaker-as.among.us",
+            Servers: [
+                {
+                    Name: "Http-1",
+                    Ip: "https://matchmaker-as.among.us",
+                    Port: 443,
+                    UseDtls: true,
+                    Players: 0,
+                    ConnectionFailures: 0,
+                },
+            ],
             TranslateName: 291,
-            UseDtls: true,
         },
         // Followed by the custom region
         {
-            $type: "DnsRegionInfo, Assembly-CSharp",
-            Fqdn: fqdn,
-            DefaultIp: ip,
-            Port: port,
+            $type: "StaticHttpRegionInfo, Assembly-CSharp",
             Name: name,
+            PingServer: fqdn,
+            Servers: [
+                {
+                    Name: "http-1",
+                    Ip: protocol + "://" + fqdn,
+                    Port: port,
+                    UseDtls: false, // As no custom key can be specified, we need to disable DTLS on custom servers.
+                    Players: 0,
+                    ConnectionFailures: 0,
```

Comment:
Leave these out and let them be replaced with defaults automatically?

Reply:
do the servers get filled in automaticallly?

Reply:
Servers? I mean `Players` and `ConnectionFailures`, they both default to `0` anyway. Either way doesn't really matter, just a nitpick (and me wanting to keep this js script to the bare minimum).